doc/Syscalls.md

```diff
@@ -156,64 +156,70 @@ in the libtock-c syscall code for more information about Yield.
-handler. Back in the context switching code, the kernel restores its registers
-from its stack. Then, using the contents of `mcause` the kernel decides why the
-application stopped executing, and if it was a syscall which syscall the app
-called. Returning the context switch reason ends the context switching process.
-
-All values for the syscall functions are passed in registers `a0-a4`. No values
-are stored to the application stack. The return value for syscalls is set in a0.
-In most syscalls the kernel will not clobber any userspace registers except for
-this return value register (`a0`). However, the `yield()` syscall results in a
-callback getting run in the app. This can clobber all caller saved registers, as
-well as the return address (`ra`) register.
+The RISC-V architecture provides very lean support for context
+switching, providing significant flexibility in software on how to
+support context switches. The hardware guarantees the following will
+happen during a context switch: when switching from kernel mode to
+user mode by calling the `mret` instruction, the PC is set to the
+value in the `mepc` CSR, and the privilege mode is set to the value in
+the `MPP` bits of the `mstatus` CSR. When switching from user mode to
+kernel mode using the `ecall` instruction, the PC of the `ecall`
+instruction is saved to the `mepc` CSR, the correct bits are set in
+the `mcause` CSR, and the privilege mode is restored to machine
+mode. The kernel can store 32 bits of state in the `mscratch` CSR.
+
+Tock handles context switching using the following process. When
+switching to userland, all register contents are saved to the kernel's
+stack. Additionally, a pointer to a per-process struct of stored
+process state and the PC of where in the kernel to resume executing
+after the process switches back to kernel mode are stored to the
+kernel's stack. Then, the PC of the process to start executing is put
+into the `mepc` CSR, the kernel stack pointer is saved in `mscratch`,
+and the previous contents of the app's registers from the per-process
+stored state struct are copied back into the registers. Then `mret` is
+called to switch to user mode and begin executing the app.
+
+An application calls a system call with the `ecall` instruction. This
+causes the trap handler to execute. The trap handler checks
+`mscratch`, and if the value is nonzero then it contains the stack
+pointer of the kernel and this trap must have happened while the
+system was executing an application. Then, the kernel stack pointer
+from `mscratch` is used to find the pointer to the stored state
+struct, and all process registers are saved. The trap handler also
+saves the process PC from the `mepc` CSR and the `mcause` CSR. It then
+loads the kernel address of where to resume the context switching code
+to `mepc` and calls `mret` to exit the trap handler. Back in the
+context switching code, the kernel restores its registers from its
```

Comment:
This whole diff seems unrelated to this PR?

Reply:
It's just a formatting/line wrapping change.